pkg/workspace/library_loader.go

```diff
@@ -116,7 +116,8 @@ func (ll *LibraryLoader) filesByAnnotation(annName structmeta.AnnotationName, lo
 				return nil, err
 			}
 
-			tplOpts := yamltemplate.MetasOpts{IgnoreUnknown: ll.templateLoaderOpts.IgnoreUnknownComments}
+			// we're only peeking at the document to determine its "type", no need to detect (yet) if contains unknowns.
+			tplOpts := yamltemplate.MetasOpts{IgnoreUnknown: true}
```

Comment:
should we move this into DocExtractor?

Reply:
Well played.

Yeah, DocExtractor being insensitive to "unknown comments" in exchange for a simpler interface there feels like a win.